parquet-common/src/main/java/org/apache/parquet/util/AutoCloseableResources.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.util;
+
+import java.io.IOException;
+
+public class AutoCloseableResources implements AutoCloseable {
```

Comment:
Why not directly work on `AutoCloseables`?

Reply:
well I tried to , but AutoCloseables just contain static functions, To me it just seems to ensure that all the passed AutoCloseables are closed. , nothing gets missed.
I cannot figure out how to use it in paradigm of 
try (    AutoCloseable resources) {
    // some operations like say 
   flush
}

Reply:
Does checking if the object to close is an impl of `Flushable` help?

Reply:
I suppose it can.. but do we want to use that check in AutoCloseables? Coz the suggested paradigm for AutoCloseable is try ( ) { }.  What right now AutoCloseables is doing, not sure why it is requiring AutoCloseable? It could well have been Closeable?

Reply:
Also it may cause issues, and possible messy code, like say we have 3 AutoCloseable Resources, having flush.. And the safest behaviour would be to flush each and then close in the end the three.
If in a single loop ,we put flush and close, it might create problems.  so to be safe, we would may be have to split into two loops, first flushables doing flush, then. closing...  Which would mean handling IOExceptions of the flushable loop too.

Reply:
I agree that calling flush in AutoCloseables may introduce more troubles if exception happens. However, adding duplicate code like this may also impose maintenance burden.